### ISSUE TITLE
Regolith 3: Unable to screen share my windows when using SWAY

### ISSUE THREAD

**Issue by tigrato** [regolith-linux/regolith-desktop#920]
**Describe the bug**

Regolith3-sway is unable to screen share any of my screens/windows in Chrome/Firefox. 

**To Reproduce**
Open a google meet video and try sharing your screen

**Expected behavior**
Regolith sway should support screen sharing

**Configuration file(s)**
Default config doesn't work.

I tried adding `XDG_CURRENT_DESKTOP=sway` to `~/.config/environment.d/xdg.conf`. With that config, I can share a single display from my computer (I can't select which one) but I loose access to Regolith settings.

**Installation Details**
 - Regolith Install Type: PPA
 - Regolith Version: 3.0
 - PPA url: [default: ppa:regolith-linux/release]
 - Host OS (for PPA):  UBUNTU 23.04
 
**Additional context**

**Comment by SoumyaRanjanPatnaik** [comment]
Try seeing the environment variable to `GNOME:sway`

**Comment by tigrato** [comment]
I tried it and it doesn't work either.the behavior is exactly the same as if it was set to `GNOME`

Regarding the `mod+c` command when `XDG_CURRENT_DESKTOP=sway`, it doesn't work and the settings doesn't show up on available applications as well.

**Comment by SoumyaRanjanPatnaik** [comment]
What is the output of the following command?
`echo $XDG_CURRENT_DESKTOP`

I have set it to `GNOME:sway` (notice the colon in between), and it works flawlessly for me. Screen sharing works, and the app launcher (`ilia`) shows me an entry for the settings app.

**Comment by tigrato** [comment]
I had to uninstall `xdg-desktop-portal-gnome`, install  slurp and set `XDG_CURRENT_DESKTOP=GNOME:sway` in order for it to work correctly.

Thank you @SoumyaRanjanPatnaik !

**Comment by SoumyaRanjanPatnaik** [comment]
Was slurp not auto installed?

**Comment by tigrato** [comment]
No :cry:

**Comment by Marcello09** [comment]
Hello, i'm also facing the same problem. 

Ubuntu 22.04. Fresh install of regolith3. Can´t screen share.
slupt is not installed by default.

I've created a config file in `~/.config/environment.d` setting `XDG_CURRENT_DESKTOP=GNOME:sway` but no success in sharing screen yet

**Comment by Marcello09** [comment]
> I had to uninstall `xdg-desktop-portal-gnome`, install slurp and set `XDG_CURRENT_DESKTOP=GNOME:sway` in order for it to work correctly.
> 
> Thank you @SoumyaRanjanPatnaik !

Doing exacly this worked for me. I had to set the environment in my `.zshrc` since putting it into `.config/enviroment.d/sway_env.conf` did not set the environment variable.

**Comment by SoumyaRanjanPatnaik** [comment]
> Doing exacly this worked for me. I had to set the environment in my .zshrc since putting it into .config/enviroment.d/sway_env.conf did not set the environment variable.

I assume that `sway_env` worked and `.zshrc` didn't (since sway doesn't import .profile, .bashrc, .zshrc, etc, during the session start). However, the startup script explicitly loads `.config/enviroment.d/sway_env.conf`. 

Glad that you managed to solve this issue though.

**Comment by lawrencegripper** [comment]
I think I found why the `sway_env` file in `environment.d` didn't work. It looks like the sway launcher loads those `environment.d` then sets the `XDG_CURRENT_DESKTOP` meaning these get overridden. 

https://github.com/regolith-linux/regolith-session/blob/859cc680413992df3085a3d7b509775920f4e7f8/usr/bin/regolith-session-wayland#L39-L45

Added a mini-PR here to fix that up, so the setting can be easily overridden.

**Comment by kgilmer** [comment]
Hi @lawrencegripper , I haven't been tracking this issue, but noticed your change coincided with some updates we're making for Regolith 3.2.  Could take a peek at the version of the Wayland session and comment if you see any concerns?

https://github.com/regolith-linux/regolith-session/blob/feat/systemd-session-merge/usr/bin/regolith-session-wayland#L49

**Comment by SoumyaRanjanPatnaik** [comment]
Can anyone facing this issue check if the `xdg-desktop-portal-regolith-wayland-config` package is installed?

**Comment by SoumyaRanjanPatnaik** [comment]
> It doesn't show up in a search for me either (using https://regolith-desktop.org/release-3_1-ubuntu-jammy-amd64 jammy InRelease )

AFAIK xdg-desktop-portal-regolith-* should support jammy. Maybe its the `InRelease` portion of the source (`https://regolith-desktop.org/release-3_1-ubuntu-jammy-amd64 jammy InRelease`). Maybe try swapping `InRelease` with `main`.

Full disclaimer I don't have a complete understanding of how apt sources work, and might be wrong about this, since you were able to upgrade to regolith 3.1

**Comment by Marcello09** [comment]
> Can anyone facing this issue check if the `xdg-desktop-portal-regolith-wayland-config` package is installed?

Just checket, it is not.
The only packages installed related to xdg-desktop are the follow

```
xdg-desktop-portal-gtk/jammy,now 1.14.0-1build1 amd64 [installed,automatic]
xdg-desktop-portal-wlr/jammy,now 0.5.0-3 amd64 [installed,automatic]
xdg-desktop-portal/jammy-updates,now 1.14.4-1ubuntu2~22.04.1 amd64 [installed,automatic]
```

**Comment by SoumyaRanjanPatnaik** [comment]
@kgilmer seems like the regolith backend for xdg-dekstop-portal isn't installed on jammy.

**Comment by kgilmer** [comment]
> @kgilmer seems like the regolith backend for xdg-dekstop-portal isn't installed on jammy.

Hmm, that package is associated via `Recommends`, here.  (The `config` package expresses a `Depends` on the root package)  I don't see any varation in the package model that would explain why this package wouldn't be installed on a specific release.  I wonder if, when apt is upgrading from one release to another, if it ignores changes to `Recommends` in the upgraded version of the package...

**Comment by vitorarins** [comment]
Hi, I also have this issue. Running on Ubuntu Jammy. Do I understand correctly that I need to install the `xdg-desktop-portal-regolith-wayland-config` package?
If so, how do I get it? Because as stated before, it doesn't appear when doing `apt search`.

my apt sources list for regolith is `deb [arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] https://regolith-desktop.org/release-3_1-ubuntu-jammy-amd64 jammy main`

**Comment by vitorarins** [comment]
Hi, I also have this issue. Running on Ubuntu Jammy. Do I understand correctly that I need to install the `xdg-desktop-portal-regolith-wayland-config` package?
If so, how do I get it? Because as stated before, it doesn't appear when doing `apt search`.

my apt sources list for regolith is `deb [arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] https://regolith-desktop.org/release-3_1-ubuntu-jammy-amd64 jammy main`

**Comment by mfisher87** [comment]
>I don't see any varation in the package model that would explain why this package wouldn't be installed on a specific release. 

@kgilmer For me, when I installed Regolith 3.2 today, that package did show up in the apt output as a recommended package. However, it's not available for me to install.

**Comment by kgilmer** [comment]
Good catch, yes the package wasn't available on jammy.  Oversight on my part.  The above PRs should address this for the 3.3 release.  If you'd like this fix earlier you'd be able to install the package from the `unstable` stage once the PRs are merged and a build is available.

**Comment by mfisher87** [comment]
:raised_hands:  Amazing, thank you Ken!

**Comment by mfisher87** [comment]
I got this working on my machine by installing `xdg-desktop-portal-regolith-wayland-config` from unstable, then setting 

```
XDG_CURRENT_DESKTOP=sway
```

Embarrassingly, I think I did it in the most wrong way possible, as setting in `~/.config/environmend.d` was having no effect; I modified `/usr/bin/regolith-session-wayland`. 

I also set `MOZ_WAYLAND_ENABLE=1` early on, I haven't tested whether that was essential.

**Comment by mfisher87** [comment]
>I am only able to share an entire screen, not individual windows like I'm used to. Is that expected?

This is expected. See: https://github.com/emersion/xdg-desktop-portal-wlr/issues/107

Maybe this will work in Regolith 3.3 based on the latest comments in that issue?

**Comment by mpareja** [comment]
[Sway 1.12](https://github.com/swaywm/sway/releases/tag/1.12) has finally been released and adds support for capturing individual windows.  Looking forward to this being available in apt repositories (Ubuntu 25.10 is currently on 1.10.1).

**Comment by SoumyaRanjanPatnaik** [comment]
Hi. This will most likely not be available till 26.10 to 27.04 I guess. Historically the `sway` package in the debian repos is delayed by upto an year. Once it gets published to upstream repos, we should probably also have it available in regolith.
